### ISSUE TITLE
Uncaught TypeError: Cannot read property 'start' of undefined

### ISSUE THREAD

**Issue by soren5** [t9md/atom-vim-mode-plus#802]
[Enter steps to reproduce:]

1. Copied text from a webpage
2. Not sure what I did here to trigger the bug
3. Pasted it using "p"

**Atom**: 1.18.0 x64
**Electron**: 1.3.15
**OS**: Ubuntu 17.04
**Thrown From**: [vim-mode-plus](https://github.com/t9md/atom-vim-mode-plus) package 0.93.0

### Stack Trace

Uncaught TypeError: Cannot read property 'start' of undefined

```
At /home/soren5/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:130

TypeError: Cannot read property 'start' of undefined
    at PutAfter.PutBefore.adjustCursorPosition (/packages/vim-mode-plus/lib/operator.coffee:570:8)
    at Function.module.exports.Emitter.simpleDispatch (/usr/share/atom/resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/usr/share/atom/resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at VimState.module.exports.VimState.emitDidFinishMutation (/packages/vim-mode-plus/lib/vim-state.coffee:161:38)
    at /packages/vim-mode-plus/node_modules/delegato/lib/delegator.js:67:61)
    at PutAfter.Operator.startMutation (/packages/vim-mode-plus/lib/operator.coffee:216:6)
    at PutAfter.Operator.execute (/packages/vim-mode-plus/lib/operator.coffee:220:6)
    at PutAfter.PutBefore.execute (/packages/vim-mode-plus/lib/operator.coffee:565:5)
    at OperationStack.execute (/packages/vim-mode-plus/lib/operation-stack.coffee:160:27)
    at OperationStack.process (/packages/vim-mode-plus/lib/operation-stack.coffee:150:8)
    at OperationStack.run (/packages/vim-mode-plus/lib/operation-stack.coffee:89:12)
    at /packages/vim-mode-plus/lib/base.coffee:368:35)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app/src/command-registry.js:265:35)
    at /usr/share/atom/resources/app/src/command-registry.js:3:65
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/usr/share/atom/resources/app/node_modules/atom-keymap/lib/keymap-manager.js:610:22)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/share/atom/resources/app/node_modules/atom-keymap/lib/keymap-manager.js:401:28)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/usr/share/atom/resources/app/src/window-event-handler.js:100:42)
    at HTMLDocument.<anonymous> (/usr/share/atom/resources/app/src/window-event-handler.js:3:65)
```

### Commands

```
     -2:35.7.0 core:cancel (input.hidden-input)
  6x -2:34.5.0 vim-mode-plus:move-left (input.hidden-input)
     -2:33.1.0 vim-mode-plus:replace-character (input.hidden-input)
     -2:32.8.0 vim-mode-plus:set-input-char-" (input.hidden-input)
  3x -2:28.9.0 vim-mode-plus:move-up (input.hidden-input)
  6x -2:26.9.0 vim-mode-plus:move-down (input.hidden-input)
     -0:34 core:save (input.hidden-input)
     -0:32.8.0 vim-mode-plus:set-count-2 (input.hidden-input)
  2x -0:31.9.0 vim-mode-plus:next-tab (input.hidden-input)
     -0:26.7.0 vim-mode-plus:move-up (input.hidden-input)
     -0:26.2.0 vim-mode-plus:move-down (input.hidden-input)
  3x -0:23.4.0 vim-mode-plus:next-tab (input.hidden-input)
  8x -0:21.9.0 vim-mode-plus:move-down (input.hidden-input)
     -0:10 vim-mode-plus:activate-blockwise-visual-mode (input.hidden-input)
     -0:08.4.0 vim-mode-plus:put-after (input.hidden-input)
     -0:02.7.0 vim-mode-plus:activate-normal-mode (input.hidden-input)
```

### Non-Core Packages

```
ariake-dark-syntax 0.1.3 
autocomplete-python 1.9.1 
vim-mode-plus 0.93.0 
```

**Comment by dylnmc** [comment]
Just a one time thing, or can you reproduce it?

I'm on arch (and GNOME desktop envir. for testing this) and can't seem to reproduce it (even when my clipboard selection is emptied with `xsel -bi <<< ""` from cmd line).

Although, I do have Electron `v1.4.16`.

**Comment by soren5** [comment]
Just a one time thing, I've tried to reproduce it but with no success. It's not a huge bother though, I only posted in case it was a common issue.


Em 28 de Jun de 2017 às 00:44, Dylan McClure <notifications@github.com> escreveu:



Just a one time thing, or can you reproduce it?

I'm on arch (and GNOME desktop envir. for testing this) and can't seem to reproduce it (even when my clipboard selection is emptied with xsel -bi <<< "" from cmd line).

Although, I do have Electron v1.4.16.

**Comment by t9md** [comment]
Thanks for reporting, I cannot say the exact situation, but can imagine this is possible situation.

E.g. in multiple-cursor operation, and two selection intersects as a result of operation, these two selection would be **merged**(then one selection is destroyed).

**Comment by rsese** [comment]
I just ran into this running from Atom master `1.27.0-dev-aa469d5e1`.  Besides the error while pasting, deleting doesn't seem to do anything (e.g when I try to delete a line with <kbd>dd</kbd>) though there are no errors in that case.  Can't reproduce in `1.26.0-beta1`.

To double check that it wasn't an issue with another community package, I reset to factory defaults and then installed vim-mode-plus again.

If you'd like me to open a new issue or have questions @t9md, please let me know (would be curious to know if you see the same issue?). 

[Enter steps to reproduce:]

After building Atom from master (I was running off of aa469d5e1), resetting to factory defaults, and installing vim-mode-plus:

1. Create a new file and type some text.
2. Yank the line you just typed <kbd>yy</kbd>
3. Paste <kbd>p</kbd>

**Atom**: 1.27.0-dev-aa469d5e1 x64
**Electron**: 1.7.11
**OS**: Mac OS X 10.12.6
**Thrown From**: [vim-mode-plus](https://github.com/t9md/atom-vim-mode-plus) package 1.32.0


### Stack Trace

Uncaught TypeError: Cannot read property 'start' of undefined

```
At /Users/rsese/.atom/packages/vim-mode-plus/lib/operation-stack.js:164

TypeError: Cannot read property 'start' of undefined
    at PutAfter.adjustCursorPosition (/packages/vim-mode-plus/lib/operator.js:732:68)
    at /packages/vim-mode-plus/lib/operator.js:705:33
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at VimState.emitDidFinishMutation (/packages/vim-mode-plus/lib/vim-state.js:179:43)
    at PutAfter.emitDidFinishMutation (/packages/vim-mode-plus/lib/base.js:299:79)
    at PutAfter.postMutate (/packages/vim-mode-plus/lib/operator.js:271:10)
    at PutAfter.execute (/packages/vim-mode-plus/lib/operator.js:287:10)
    at PutAfter.execute (/packages/vim-mode-plus/lib/operator.js:687:16)
    at OperationStack.execute (/packages/vim-mode-plus/lib/operation-stack.js:200:33)
    at OperationStack.process (/packages/vim-mode-plus/lib/operation-stack.js:193:12)
    at OperationStack.run (/packages/vim-mode-plus/lib/operation-stack.js:105:14)
    at /packages/vim-mode-plus/lib/vim-state.js:70:33)
    at CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app/src/command-registry.js:384:49)
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:621:22)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:412:28)
    at WindowEventHandler.handleDocumentKeyEvent (/Applications/Atom.app/Contents/Resources/app/src/window-event-handler.js:110:40)
```

### Commands

```
     -0:57.3.0 vim-mode-plus:activate-insert-mode (input.hidden-input)
     -0:02.6.0 vim-mode-plus:activate-normal-mode (input.hidden-input)
  2x -0:02.2.0 vim-mode-plus:yank (input.hidden-input)
     -0:01.8.0 vim-mode-plus:put-after (input.hidden-input)
```

### Non-Core Packages

```
vim-mode-plus 1.32.0 
```

**Comment by t9md** [comment]
@rsese Thanks, will investigate further but I believe this will be fixed when I fixed #1058.
